src/module-server/plugins/npm-plugin.ts

```diff
@@ -0,0 +1,229 @@
+import { dirname, join, normalize, posix } from 'path';
+import type { Plugin, RollupCache } from 'rollup';
+import { rollup } from 'rollup';
+import { existsSync, promises as fs } from 'fs';
+import { resolve, legacy as resolveLegacy } from 'resolve.exports';
+import commonjs from '@rollup/plugin-commonjs';
+import { processGlobalPlugin } from './process-global-plugin';
+import * as esbuild from 'esbuild';
+import { parse } from 'cjs-module-lexer';
+import MagicString from 'magic-string';
+import { fileURLToPath } from 'url';
+
+// This is the folder that Pleasantest is installed in (e.g. <something>/node_modules/pleasantest)
+const installFolder = dirname(dirname(dirname(fileURLToPath(import.meta.url))));
+// Something like <something>/node_modules/pleasantest/.cache
+const cacheDir = join(installFolder, '.cache');
+
+const npmTranspileCache = new Map<string, string>();
+const setInCache = (cachePath: string, code: string) => {
+  npmTranspileCache.set(cachePath, code);
+  fs.mkdir(dirname(cachePath), { recursive: true }).then(() =>
+    fs.writeFile(cachePath, code),
+  );
+};
+
+const getFromCache = async (cachePath: string) => {
+  return (
+    npmTranspileCache.get(cachePath) ||
+    fs
+      .readFile(cachePath, 'utf8')
+      .catch(() => {})
```

Comment:
Is this empty function in here by design? If so, can you leave a comment explaining why? 🤔